src/main/kotlin/com/exactpro/th2/codec/fixng/ByteBufUtil.kt

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2023-2024 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.codec.fixng
+
+import io.netty.buffer.ByteBuf
+import java.nio.charset.Charset
+
+private const val SEP_BYTE = '='.code.toByte()
+private const val SOH_BYTE = 1.toByte()
+private const val DIGIT_0 = '0'.code.toByte()
+private const val DIGIT_9 = '9'.code.toByte()
+
+private fun Int.getDigitCount(): Int = when {
+    this < 10 -> 1
+    this < 100 -> 2
+    this < 1000 -> 3
+    this < 10000 -> 4
+    this < 100000 -> 5
+    this < 1000000 -> 6
+    this < 10000000 -> 7
+    this < 100000000 -> 8
+    this < 1000000000 -> 9
+    else -> 10
+}
+
+private fun ByteBuf.printInt(sourceValue: Int, digits: Int = sourceValue.getDigitCount()): ByteBuf = apply {
+    var value = sourceValue
+
+    ensureWritable(digits)
+
+    repeat(digits) { index ->
+        setByte(digits - index - 1 + writerIndex(), value % 10 + DIGIT_0)
+        value /= 10
+    }
+
+    writerIndex(writerIndex() + digits)
+}
+
+fun ByteBuf.readTag(): Int {
+    val offset = readerIndex()
+    var tag = 0
+
+    while (isReadable) {
+        val byte = readByte()
+
+        if (byte in DIGIT_0..DIGIT_9) {
```

Comment:
suggestion: `DIGIT_0 <= byte && byte <= DIGIT_9`

Reply:
Just for your information @Nikita-Smirnov-Exactpro: the code from your suggestion is equivalent to the original code. The `Range` object won't be created here

Reply:
Thank you for notice. Decoded from byte code 
```
         byte var3 = $this$readTag.readByte();
         if (48 <= var3) {
            if (57 >= var3) {
               tag = tag * 10 + var3 - 48;
               if (tag >= 0) {
                  continue;
               }
            }
         }
         ```